From 64eb0b45bdd41c798706d5a877d3888ee64fb01c Mon Sep 17 00:00:00 2001
From: Lim Siew Hoon <siew.hoon.lim@intel.com>
Date: Mon, 31 Jul 2023 18:56:20 +0800
Subject: [PATCH 45/77] video: add GST_VIDEO_DMA_DRM_CAPS_MAKE

It is requried this macro because it doesn't define format field, neither
drm-format which is suppossed to be generated in run-time.

Part-of: <https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/4981>

Backport from upstream:
https://gitlab.freedesktop.org/gstreamer/gstreamer/-/commit/8b38614173e09f9acea00a906882da169bfe9e91

Signed-off-by: Lim Siew Hoon <siew.hoon.lim@intel.com>
---
 gst-libs/gst/video/video-info-dma.h | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/gst-libs/gst/video/video-info-dma.h b/gst-libs/gst/video/video-info-dma.h
index 15e6a87c..a0f9246d 100644
--- a/gst-libs/gst/video/video-info-dma.h
+++ b/gst-libs/gst/video/video-info-dma.h
@@ -25,6 +25,22 @@
 
 G_BEGIN_DECLS
 
+/**
+ * GST_VIDEO_DMA_DRM_CAPS_MAKE:
+ *
+ * Generic caps string for video wit DMABuf(GST_CAPS_FEATURE_MEMORY_DMABUF)
+ * feature, for use in pad templates. As drm-format is supposed to be defined
+ * at run-time it's not predefined here.
+ *
+ * Since: 1.24
+ */
+#define GST_VIDEO_DMA_DRM_CAPS_MAKE                                     \
+    "video/x-raw(memory:DMABuf), "                                      \
+    "format = (string) DMA_DRM, "                                       \
+    "width = " GST_VIDEO_SIZE_RANGE ", "                                \
+    "height = " GST_VIDEO_SIZE_RANGE ", "                               \
+    "framerate = " GST_VIDEO_FPS_RANGE
+
 typedef struct _GstVideoInfoDmaDrm GstVideoInfoDmaDrm;
 
 /**
-- 
2.40.1

